packages/experiments/src/components/Toggle/Toggle.types.ts

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { IComponent, IStyleableComponentProps } from '../../Foundation';
+import { IKeytipProps } from 'office-ui-fabric-react/lib/Keytip';
+import { IStyle } from 'office-ui-fabric-react/lib/Styling';
+import { IComponentAs, IRefObject } from '../../Utilities';
+
+export type IToggleComponent = IComponent<IToggleProps, IToggleViewProps, IToggleStyles>;
+
+/**
+ * Toggle component props.
+ */
+export interface IToggleProps extends IStyleableComponentProps<IToggleViewProps, IToggleStyles> {
+  /**
+   * Render the root element as another type.
+   */
+  as?: IComponentAs<React.HTMLAttributes<HTMLElement>>;
+
+  /**
+   * Optional callback to access the IToggleComponent interface. Use this instead of ref for accessing
+   * the public methods and properties of the component.
+   */
+  componentRef?: IRefObject<IToggleComponent>;
```

Comment:
So, should I just take out `componentRef`? I did have to change one test because `componentRef` was working weird, basically checking the same thing without using it.

Originally, the test was:
```ts
it('doesn't trigger onSubmit when placed inside a form', () => {
    let component: any;
    const onSubmit = sinon.spy();

    const wrapper = mount(
      <form
        action="#"
        // tslint:disable-next-line:jsx-no-lambda
        onSubmit={e => {
          onSubmit();
          e.preventDefault();
        }}
      >
        <Toggle
          // tslint:disable-next-line:jsx-no-lambda
          componentRef={ref => (component = ref)}
          label="Label"
        />
      </form>
    );
    const button: any = wrapper.find('button');
    // simulate to change toggle state
    button.simulate('click');
    expect((component as React.Component<any, any>).state.checked).toEqual(true);
    expect(onSubmit.called).toEqual(false);
  });
```

And I had to change it to:
```ts
it('doesn't trigger onSubmit when placed inside a form', () => {
    const onSubmit = sinon.spy();

    const wrapper = mount(
      <form
        action="#"
        // tslint:disable-next-line:jsx-no-lambda
        onSubmit={e => {
          onSubmit();
          e.preventDefault();
        }}
      >
        <Toggle
          // tslint:disable-next-line:jsx-no-lambda
          label="Label"
        />
      </form>
    );
    const button: any = wrapper.find('button');

    // simulate to change toggle state
    button.simulate('click');

    expect(
      button
        .first()
        .getDOMNode()
        .getAttribute('aria-checked')
    ).toEqual('true');
    expect(onSubmit.called).toEqual(false);
  });
```

Just wanted to really double check if it is still doing its intended purpose.

Reply:
`componentRef` is supposed to be set to point to the component's instance, but `BaseComponent` handles that. Right now, nothing is actually setting that value in the new `Toggle`, so it wouldn't be usable.

I would revert the test and comment it out until we resolve out we deal with componentRef. Also consider temporarily commenting out `componentRef` prop unless we can resolve it in this PR.

Did you have to change any other tests?

Reply:
Sounds good, will revert and comment the test for the moment.

Hopefully, this was the only test I had to change.

Reply:
Unless @dzearing objects, I think we should copy over `BaseComponent` to where `BaseState` resides in experiments and minify it just so that it contains the `componentRef` logic as part of this PR. I'm still thinking of names (`BaseComponentMin`? something better?) but once we can promote it we can have the existing `BaseComponent` extend it.

Reply:
Also invalid type similar to Button/CS (sorry for not noticing the wrong type before.)